napalm_junos/junos.py

```diff
@@ -1763,20 +1763,49 @@ def get_optics(self):
         optics_table.get()
         optics_items = optics_table.items()
 
-        # Formatting data into return data structure
-        optics_detail = {}
+        # Format data inserting lane 0 for single lane optics
+        # Format: [(iface, lane, [optical_values]), ...]
+        optics_items_with_lane = []
         for intf_optic_item in optics_items:
+            a = list(intf_optic_item)
+            a.insert(1, u"0")
+            new_intf_optic_item = tuple(a)
```

Comment:
Can you please explain these three lines @ckishimo? (1770, 1771, 1772)

Reply:
Sure. I'm just inserting the lane 0 for each element of optics_items list, so when later getting the information for 40G/100G optics, all items will have a lane and the format will be:
```
[ ( interface_name,  lane_X,  [optical_values_for_lane_X] ) ]
```
So for instance the first element of the list being:
```
('xe-0/0/0', [('output_power', '- Inf'), ('laser_bias_current', '38.346'), ('input_power', '- Inf')]),
```
it will become::
```
('xe-0/0/0', u'0', [('output_power', '- Inf'), ('laser_bias_current', '38.346'), ('input_power', '- Inf')]),
```
Then later the list new_optics_40Gitems will have the information for all optics 10G/40G/100G and will be like:
```
('xe-0/0/0', u'0', [('output_power', '- Inf'), ('laser_bias_current', '38.346'), ('input_power', '- Inf')]),
('et-0/0/49', u'0', [('output_power', '-1.30'), ('laser_bias_current', '6.987'), ('input_power', '-1.24')]),
('et-0/0/49', u'1', [('output_power', '-0.96'), ('laser_bias_current', '6.791'), ('input_power', '-1.19')]),
('et-0/0/49', u'2', [('output_power', '-1.08'), ('laser_bias_current', '6.791'), ('input_power', '-1.00')]),
('et-0/0/49', u'3', [('output_power', '-1.07'), ('laser_bias_current', '6.955'), ('input_power', '-1.17')]),
```
Once all information is in the same format we can iterate in the last loop...

if that was not clear I'm sure there is a better way of coding that :smiley: I'm still newbie ... Let me know. Thanks @mirceaulinic !

Reply:
Okay, that makes sense - thanks @ckishimo.
As this is not immediately obvious, can you please insert some comments to explain this (what you wrote above works too, eventually a shorter version).

Reply:
And maybe rename `a` to `temp_list` or something more descriptive.